### PR TITLE
meter publishing result

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

Report publishing success or failure.

## Motivation and Context

We had a bad incident some time ago that publishing didn't work and Styx retried 420 times and that delayed workflow execution badly. We want to have a metric that we can monitor and take early action when it happens again.

## Have you tested this? If so, how?
 

Unit test.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [x] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [x] This PR has been incorporated in release note for the coming version
- [x] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java [hunk 1]
```diff
@@ -40,16 +46,28 @@
...
+  private final Retrier retrier;
   private final Publisher publisher;
+  private final Stats stats;
 
-  public PublisherHandler(Publisher publisher) {
+  public PublisherHandler(Publisher publisher, Stats stats) {
+    this(publisher, stats,
+        Retrier.builder()
+            .errorMessage("publish deploy event")
+            .retryDelay(Duration.ofSeconds(1))
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java]: Exponential backoff?

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java]: Could be, but not in this PR I think.

Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java]: If we really think it is important to send this event, then we should persist it to durable storage.

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java]: I don't actually know the initial requirement of ITGC.

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java]: This has been very reliable, meaning ~100% in the past years. Not sure it is really worth persisting that includes more moving parts.

**Comment by danielnorberg** [comment]
👍

File: styx-common/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 2]
```diff
@@ -148,6 +148,13 @@ public boolean isInstance(Metric metric) {
       .tagged("what", "workflow-consumer-error-rate")
       .tagged("unit", "error");
 
+  static final MetricId PUBLISHING_RATE = BASE
+      .tagged("what", "publishing-rate");
+
+  static final MetricId PUBLISHING_ERROR_RATE = BASE
+      .tagged("what", "publishing-error-rate")
+      .tagged("unit", "error");
```
Comment (ulzha) [inline styx-common/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: BTW I guess it's a wrong unit for a rate.

Comment (honnix) [reply styx-common/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: I think we once had a PR to remove all those faulty units. No idea why they make their way back. :(

**Pull request merged**